docs/guides/upgrade-apps-without-downtime.md

```diff
@@ -0,0 +1,182 @@
+
+During an upgrade of an application, Kubernetes starts the Pods of the new version and brings down the old ones. It also
+deletes and creates the corresponding Endpoints. NGINX Kubernetes Gateway sees the changes to the Endpoints by watching
+for changes to the corresponding [EndpointSlices][endpoint-slices].
+
+[endpoint-slices]:https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/
+
+In NGINX configuration, a Service is represented as an [upstream][upstream], and an Endpoint as an
+[upstream server][upstream-server].
+
+[upstream]:https://nginx.org/en/docs/http/ngx_http_upstream_module.html#upstream
+
+[upstream-server]:https://nginx.org/en/docs/http/ngx_http_upstream_module.html#server
+
+Let's consider two cases:
+
+- If an Endpoint is added, NGINX Kubernetes Gateway adds an upstream server to NGINX that corresponds to the Endpoint,
+  and then reloads NGINX. After that, NGINX will start proxying traffic to that Endpoint.
+- If an Endpoint is removed, NGINX Kubernetes Gateway removes the corresponding upstream server from NGINX. After
+  a reload, NGINX will stop proxying traffic to it. However, it will finish proxying any pending requests to that
+  server,
+  which prevents downtime.
+
+As a result, as long you have more than one ready Endpoint, the clients should not experience any downtime during
+an upgrade.
+
+> It is a good practice to configure a [Readiness probe][readiness-probe] in the Deployment so that a pod can advertise
+> when it is ready to receive traffic. Note that NGINX Kubernetes Gateway will not add any Endpoint to NGINX that is not
+> ready.
+
+[readiness-probe]:https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
+
+The upgrade methods from the next sections will change upstream servers in NGINX.
+
+## Rolling Deployment Upgrade
+
+To start a [rolling Deployment upgrade][rolling-upgrade], you update the Deployment to use the new version tag of
+the application. As a result, Kubernetes terminates the Pods with the old version and create new ones. By default,
+Kubernetes also ensures that some number of pods always stay available during the upgrade.
+
+[rolling-upgrade]:https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#rolling-update-deployment
+
+Such an upgrade will add new upstream servers to NGINX and remove the old ones. As long as the number
+of Pods (ready Endpoints) during an upgrade does not reach zero, NGINX will be able to proxy traffic, and thus prevent
+any downtime.
+
+To use this method, it is not necessary to update the HTTPRoute.
+
+## Blue-Green Deployments
+
+With this method, you deploy a new version of the application (blue version) as a separate Deployment,
+while the old version (green) keeps running and handling client traffic. Next, you switch the traffic from the
+green version to the blue. If the blue works as expected, you terminate the green or switch the traffic back to it
+otherwise.
```

Comment:
For readability, can we say something like... "If the blue application works as expected, the green application should be terminated. If the blue application does not work as expected, the traffic should be switched back to the green application."